RevisionNotes/MIT_Lec12.md

```diff
@@ -0,0 +1,222 @@
+`x` is basically the potential at each node.
+
+`Ax = 0`
+
+![Nullspace of A](Images/Nullspace_lec12.PNG)
+
+![Potential Difference](Images/PotentialDiff_lec12.PNG)
+
+`Ax` denotes the Potential Difference between nodes.
+
+It is 0 when all the potentials are the same.
+
+So nullspace includes all vectors of the form
+`c(1, 1, 1, 1)`
+
+So the <b>dimension of nullspace = `n - r` = 1</b>
+
+Current doesn't flow when potential difference is 0.
+
+We can raise or lower all voltages by the same amount `c`, without changing the differences. There is an "arbitrary constant" in the voltages. 
+
+The nullspace disappears when we fix x<sub>4</sub> = 0. The unknown x<sub>4</sub> is removed and so is the fourth column of A (the column multiplied x<sub>4</sub>). Basically, node 4 has been "grounded." So only zero vector is in the nullspace now.
+
+#### <u>The Row Space</u>
+
+The row space contains all combinations of the five rows.
+
+<b>Dimension of rowspace = `r` = 3</b>
+
+We got this from elimination.
+
+How do we know if a vector is in the rowspace?
+
+If it is perpendicular to (1, 1, 1, 1) in the nullspace. (Orthogonality)
+
+#### <u>The Column Space</u>
+
+The column space contains all combinations of the four columns. We expect three independent columns, since there were three independent rows. The first three columns of A are independent (so are any three).
+
+So, <b>dimension of column space = `r` = 3</b>
+
+How can we tell if a particular vector b is in the column space of an incidence matrix?
+
+The components of b should add to 0 around every loop. Why?
+
+`Ax` gives the vector of potential differences.
+
+`Ax = b`
+
+If we add differences around a closed loop in the graph, they cancel to leave zero.
+
+For example, let's take the potential differences for this loop.
+
+![Loop](Images/ColumnSpace_lec12.PNG)
+
+Here, `(x`<sub>`2`</sub>` - x`<sub>`1`</sub>`) + (x`<sub>`3`</sub>` - x`<sub>`2`</sub>`) - (x`<sub>`3`</sub>` - x`<sub>`1`</sub>`) = 0`
+
+These correspond to the components of b:
+
+Therefore, `b`<sub>`1`</sub>` + b`<sub>`2`</sub>` - b`<sub>`3`</sub>` = 0`
+
+So a quick way to check if a vector is in the column space is to check whether the components add up to 0 around a loop (considering directions).
+
+#### <u>The Left Nullspace</u>
+
+The left nullspace contains the solutions to `A`<sup>`T`</sup>`y = 0`.
+
+<b>Dimension of left nullspace = `m - r` = 2</b>
+
+![Left Nullspace](Images/LeftNullspace_lec12.PNG)
+
+![Current](Images/Current_lec12.PNG)
+
+Here, `y` is the current along each edge.
+
+Thus, the net flow at each node should be zero.
+
+We can find the basis by assuming current (`y`) through one edge and solving such that there is no charge accumulation.
+
+What are solutions to `A`<sup>`T`</sup>`y = 0`?
+
+Currents which balance themselves. Every loop current is a solution.
+
+![Current Loops](Images/Loops_lec12.PNG)
+
+Here, we have 2 small independent loops. The big loop of 1-2-3-4 is basically the sum of these two loops.
```

Comment:
Do you mean loop 1-2-5-4 ?

Reply:
I initially wrote it in terms of nodes. But I think 1-2-5-4 in terms of edges would be better. Corrected now.

Reply:
I have changed it to 1-2-3-4-1